addons/mrp/static/src/js/cost_structure_report.js

```diff
@@ -0,0 +1,64 @@
+odoo.define('report.bom_cost_structure', function (require) {
+'use strict';
+var core = require('web.core');
+var ReportAction = core.action_registry.get('report.client_action');
+
+
+var BomReportAction = ReportAction.extend({
+    init: function (parent, action, options){
+        var option = _.extend({}, options, {
+                report_url: '/report/html/' + action.report_name + '/' + action.context.active_id,
+                report_name: action.report_name,
+                report_file: action.report_file,
+                data: action.data,
+                context: action.context,
+                name: action.name,
+                display_name: action.display_name,
+            });
+        this._super(parent, action, option);
+    },
+    openM2ORecord: function (){
+        var action = {
+            'type': 'ir.actions.act_window',
+            'view_type': 'form',
+            'view_mode': 'form',
+            'target': 'current',
+            'res_id': Number($(this).attr('res-id')),
+            'res_model': $(this).attr('res-model'),
+            'views': [[false, 'form']],
+        };
+        window.parent.postMessage({
+            'message': 'report:do_action',
+            'action': action,
+        }, window.origin);
```

Comment:
this should be implemented in the report client action directly

Reply:
isn't it already? [see here](https://github.com/odoo/odoo/blob/11.0/addons/web/static/src/js/report/report.editor.js#L71)

Reply:
@sle-odoo  report.editor.js only load for [editable reports](https://github.com/odoo/odoo/blob/master/addons/web/views/report_templates.xml#L32)